lib/extractor/layer.ts

```diff
@@ -53,6 +55,70 @@ export async function extractDockerArchive(
   });
 }
 
+export async function extractOciArchive(
+  ociArchiveFilesystemPath: string,
+  extractActions: ExtractAction[],
+): Promise<{
+  layers: ExtractedLayers[];
+  manifest: OciArchiveManifest;
+}> {
+  return new Promise((resolve, reject) => {
+    const tarExtractor: Extract = extract();
+
+    const layers: Record<string, ExtractedLayers> = {};
+    const manifests: OciArchiveManifest[] = [];
+    let imageIndex;
+
+    tarExtractor.on("entry", async (header, stream, next) => {
+      if (header.type === "file") {
+        if (isImageIndexFile(header.name)) {
+          imageIndex = await streamToJson<OciImageIndex>(stream);
+        } else {
+          const streamCopy1 = new PassThrough();
+          const streamCopy2 = new PassThrough();
+          stream.pipe(streamCopy1);
+          stream.pipe(streamCopy2);
+
+          const promises = [
+            streamToJson(streamCopy1).catch(() => undefined),
```

Comment:
why do we have the `.catch` instead of having a try-catch around the `Promise.all`?

Reply:
Because you want every copy to either succeed or fail, having a try-catch on the list of promises will fail all of them if one of them fails

Reply:
And when we catch an error we return `undefined`, which means that noting was read